### DISCUSSION TITLE
Lilygo T-Display S2 (ESP32S2) and Rotation

### DISCUSSION THREAD

**Discussion by justdudeone** [Bodmer/TFT_eSPI#2855] [General]
Hello,
Suppose those questions for professional is very simple, still asking for help.
1. Why my current MCU board Lilygo T-Display-S2 (ESP32S2) v.1.1 works with TFT_eSPI library only from that repository: https://github.com/Xinyuan-LilyGO/LilyGo-T-Display-S2 ??? If I download TFT_eSPI library from this original repository and use correct User_Setup (Setup71_ESP32_S2_ST7789.h from S2 library or from this newest library) display does not work?! What is the difference between those TFT_eSPI libraries, why they are not update-able??? :(
2. How to fast rotate display info from default vertical position to horizontal landscape orientation or to 270* (as it possible with Adafruit GFX)?
3. If I don't use any graphics, just simple colored text - do I really need TFT_eSPI? Or maybe better use Adafruit GFX, will it take less MCU memory or the same memory space?

**Comment by Bodmer** [comment]
Setup 71 is for a generic display. To adapt it for the Liliygo T2 you will need to create a new setup or alter it to be:
```
#define ST7789_DRIVER     // Configure all registers

#define TFT_WIDTH  135
#define TFT_HEIGHT 240

#define CGRAM_OFFSET      // Library will add offsets required

#define TFT_CS   34
#define TFT_MOSI 35
#define TFT_SCLK 36
#define TFT_MISO 4

#define TFT_DC   37
#define TFT_RST  38
#define TFT_BL  33

#define LOAD_GLCD
#define LOAD_FONT2
#define LOAD_FONT4
#define LOAD_FONT6
#define LOAD_FONT7
#define LOAD_FONT8
#define LOAD_GFXFF

#define SMOOTH_FONT

#define SPI_FREQUENCY   27000000
```

**Reply by justdudeone** [reply]
Have you read my post?
Setup71 WORKS from S2 repository AND from YOUR LAST release... BUT only with TFT_eSPI from S2 repository!
